components/display/tm1638.rst

```diff
@@ -0,0 +1,189 @@
+          it.printf(0, "%.1f", id(my_sensor).state);
+          // Result: "42.1 " (the dot will appear on the "2" segment)
+
+          // Overwrite the previous content with blank
+          it.print("    ");
+          // Print a right-padded sensor value with 0 digits after the decimal
+          it.printf("S%3.0f", id(my_sensor).state);
+          // Result: "S 42"
+
+          // Print the current time
+          it.strftime("%H.%M");
+          // Result for 10:06:42 -> "10:06" on a display with : and "10.06" on a display with .
+
+Please see :ref:`display-printf` for a quick introduction into the ``printf`` formatting rules and
+:ref:`display-strftime` for an introduction into the ``strftime`` time formatting.
+
+Creating a digital clock
+************************
+
+The following example creates a typical digital clock with the ``:`` colon flashing every second.
+
+
+.. code-block:: yaml
+
+
+    time:
+      - platform: homeassistant
+        id: homeassistant_time
+
+    display:
+      platform: tm1638
+      clk_pin: 18
+      dio_pin: 23
+      stb_pin: 5
+      update_interval: 500ms
+      lambda: |-
+          static int i = 0;
+          i++;
+          if ((i % 2) == 0)
+            it.strftime("%H.%M", id(homeassistant_time).now());
+          else
+            it.strftime("%H%M", id(homeassistant_time).now());
+
+
+LEDs
+----------------
+The TM1638 7 Segment Display Keypad & LED Module has 8 LED lights across the top or middle of the display, depending on the board.  These LEDs can be exposed as :doc:`switches </components/switch/index>` or :doc:`outputs </components/output/index>` with the led property set between 0 and 7.  The board has the LEDs labelled 1-8, but the leds use zero based addressing.   The LED labelled "1" is LED 0 in the configuration.
+
+.. code-block:: yaml
+
+    switch:                 
+      - platform: tm1638
+        id: TM1638Led1
+        led: 0
+        
+    output:
+      - platform: tm1638
+        id: TM1638Led2
+        led: 1
+    
+
+
+Buttons
+----------------
+The TM1638 7 Segment Display Keypad & LED Module has 8 buttons across the top or middle of the display, depending on the board.  These buttons are exposed as binary sensor components with the key property set between 0 and 7,  They can be accessed via the binary sensor component configuration.  As with the LEDs the buttons are labeled 1-8 on the board, but use zero based addressing.  Therefore the button labelled "1" is key 0 in the configuration.   See :doc:`/components/binary_sensor/index`
+
+.. code-block:: yaml
+
+    binary_sensor:
+      - platform: tm1638
+        name: "TM1638 Button 1"
+        id: TM1638BUtton1
+        key: 0
+        filters:
+          - delayed_on: 10ms
+        on_press:
+         then:
+            - switch.turn_on: TM1638Led1
+        on_release:
+         then:
+            - switch.turn_off: TM1638Led1
+            
+    switch:            
```

Comment:
This isn't correct.  I assume this was supposed to be a second binary_sensor.

Reply:
Yep,  corrected it.  Thanks for the catch.